sphinxcontrib/spelling/directive.py

```diff
@@ -27,18 +37,20 @@ class SpellingDirective(rst.Directive):
     def run(self):
         env = self.state.document.settings.env
 
-        # Initialize the per-document good words list
-        if not hasattr(env, 'spelling_document_words'):
-            env.spelling_document_words = collections.defaultdict(list)
-
         good_words = []
         for entry in self.content:
             if not entry:
                 continue
             good_words.extend(entry.split())
         if good_words:
-            logger.debug('Extending local dictionary for %s with %s',
-                         env.docname, good_words)
-            env.spelling_document_words[env.docname].extend(good_words)
+            add_good_words_to_document(env, env.docname, good_words)
 
         return []
+
+
+class LegacySpellingDirective(SpellingDirective):
+
+    def run(self):
+        logger.info('direct use of the spelling directive is deprecated, '
```

Comment:
@dhellmann wow! This happened two years ago and I'm only noticing it now!

Time to make this an actual Sphinx warning?